native/src/main/java/io/ballerina/observe/trace/zipkin/ZipkinTracerProvider.java

```diff
@@ -0,0 +1,106 @@
+ * under the License.
+ */
+package io.ballerina.observe.trace.zipkin;
+
+import io.ballerina.observe.trace.zipkin.sampler.RateLimitingSampler;
+import io.ballerina.runtime.api.values.BDecimal;
+import io.ballerina.runtime.api.values.BString;
+import io.ballerina.runtime.observability.tracer.spi.TracerProvider;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.propagation.ContextPropagators;
+import io.opentelemetry.exporter.zipkin.ZipkinSpanExporter;
+import io.opentelemetry.extension.trace.propagation.B3Propagator;
+import io.opentelemetry.sdk.resources.Resource;
+import io.opentelemetry.sdk.trace.SdkTracerProvider;
+import io.opentelemetry.sdk.trace.SdkTracerProviderBuilder;
+import io.opentelemetry.sdk.trace.export.BatchSpanProcessor;
+import io.opentelemetry.sdk.trace.samplers.Sampler;
+
+import java.io.PrintStream;
+import java.util.concurrent.TimeUnit;
+
+import static io.opentelemetry.semconv.resource.attributes.ResourceAttributes.SERVICE_NAME;
+
+/**
+ * This is the Zipkin tracing extension class for {@link TracerProvider}.
+ */
+public class ZipkinTracerProvider implements TracerProvider {
+    private static final String TRACER_NAME = "zipkin";
+    private static final PrintStream console = System.out;
+
+    static SdkTracerProviderBuilder tracerProviderBuilder;
+
+    @Override
+    public String getName() {
+        return TRACER_NAME;
+    }
+
+    @Override
+    public void init() {    // Do Nothing
+    }
+
+    public static void initializeConfigurations(BString agentHostname, int agentPort, BString samplerType,
+                                                BDecimal samplerParam, int reporterFlushInterval,
+                                                int reporterBufferSize) {
+
+        String reporterEndpoint = "http://" + agentHostname + ":" + agentPort + "/api/v2/spans";
+
+        ZipkinSpanExporter exporter = ZipkinSpanExporter.builder().setEndpoint(reporterEndpoint).build();
+
+        tracerProviderBuilder = SdkTracerProvider.builder()
+                .addSpanProcessor(BatchSpanProcessor
+                        .builder(exporter)
+                        .setMaxExportBatchSize(reporterBufferSize)
+                        .setExporterTimeout(reporterFlushInterval, TimeUnit.MILLISECONDS)
+                        .build());
+
+        tracerProviderBuilder.setSampler(selectSampler(samplerType, samplerParam));
+
+        console.println("ballerina: started publishing traces to Zipkin on " + reporterEndpoint);
+    }
+
+    private static Sampler selectSampler(BString samplerType, BDecimal samplerParam) {
+        switch (samplerType.getValue()) {
+            default:
+            case "const":
+                if (samplerParam.value().intValue() == 0) {
+                    return Sampler.alwaysOff();
+                } else {
+                    return Sampler.alwaysOn();
+                }
+            case "probabilistic":
```

Comment:
Lets make these values constants like `RateLimitingSampler.TYPE`

Reply:
Addressed the suggestions